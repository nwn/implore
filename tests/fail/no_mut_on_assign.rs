use implore::impl_op;

#[derive(Copy, Clone, PartialEq, Eq, Debug)]
struct X;

#[impl_op]
fn add_assign(x: &X, y: u32) {}

fn main() {}
