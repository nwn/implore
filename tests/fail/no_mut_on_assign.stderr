error: the first operand of `add_assign` must be a mutable reference
 --> $DIR/no_mut_on_assign.rs:7:18
  |
7 | fn add_assign(x: &X, y: u32) {}
  |                  ^^
